fcrepo-persistence-api/src/main/java/org/fcrepo/persistence/api/PersistentStorage.java

```diff
@@ -17,10 +17,57 @@
  */
 package org.fcrepo.persistence.api;
 
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.ws.rs.core.MediaType;
+
+import org.fcrepo.kernel.api.RdfStream;
+import org.fcrepo.kernel.api.models.FedoraResource;
+
 /**
  * An interface that mediates CRUD operations to and from persistence storage.
  *
  * @author dbernstein
+ * @author whikloj
+ * @since 2019-09-11
  */
 public interface PersistentStorage {
-}
\ No newline at end of file
+
+    /**
+     * Locate an existing resource on disk.
+     *
+     * @param identifier The id of the resource.
+     * @return the FedoraResource
+     */
+    public FedoraResource findResource(final String identifier);
+
+    /**
+     * Save/Update a RdfSource to storage.
+     *
+     * @param identifier The identifier of the resource.
+     * @param content The RdfStream of content.
+     * @param informationHeader The information required for headers or other Fedora actions.
+     */
+    public void saveRdfSource(final String identifier, final RdfStream content,
+            final Map<String, String> informationHeader);
+
+    /**
+     * Save/Update a NonRdfSource to storage.
+     *
+     * @param identifier The identifier of the resource.
+     * @param content The binary stream.
+     * @param informationHeaders The information required for headers or other Fedora actions.
+     * @param contentType The mime-type.
+     */
+    public void saveNonRdfSource(final String identifier, final InputStream content,
```

Comment:
Maybe its fine, but external binaries are non-RDFSources as well, which would not have a content stream. I am guessing that informationHeaders would be used to pass in the external binary information?

Reply:
Is the current way we deal with NonRdfSources good for external content? If there is anything that would be better if done differently, now would be the time to change it and you have the most experience with that part (IMHO).

Reply:
The NonRDFSource contains the URI of the external binary or the content of a non-external binary. It would probably be cleaner if all of the properties that are provided during a non-rdf source PUT or POST were kept on the Non-RDF source node itself, rather than being pushed off to the Description. Neither LDP or fcrepo require a binary to have a description. But at this point I don't know what the internal structure of objects will be in fcrepo6, its possible these distinctions might not be meaningful. I believe at some point there was discussion of holding fedora specific details about objects separate from the RDF of the object, to accommodate overlaying fedora on existing OCFL without modifying existing files. So that might impact this situation.

The other major complicating factor with binaries that leads to confusion is the single subject restriction, which resulted in handling of triples for binary descriptions becoming rather convoluted. If the single subject restriction is lifted, that will probably be a bit simpler.